userspace/libscap/examples/01-open/test.c

```diff
@@ -336,6 +350,118 @@ void print_parameter(int16_t num_param)
 	valptr += len;
 }
 
+void print_ordered_syscalls(char string_vector[SYSCALL_TABLE_SIZE][SYSCALL_NAME_MAX_SIZE], int dim)
+{
+	char temp[SYSCALL_NAME_MAX_SIZE];
+
+	/* storing strings in the lexicographical order */
+	for(int i = 0; i < dim; ++i)
+	{
+		for(int j = i + 1; j < dim; ++j)
+		{
+			/* swapping strings if they are not in the lexicographical order */
+			if(strcmp(string_vector[i], string_vector[j]) > 0)
+			{
+				strcpy(temp, string_vector[i]);
+				strcpy(string_vector[i], string_vector[j]);
+				strcpy(string_vector[j], temp);
+			}
+		}
+	}
+
+	printf("\nSyscalls in the lexicographical order: \n");
+	for(int i = 0; i < dim; i++)
+	{
+		printf("[%d] %s\n", i, string_vector[i]);
+	}
+	printf("Interesting syscalls: %d\n", dim);
+}
+
+/* This are the real interesting syscalls that we want to support in the new probe.
+ * - all syscalls associated with events of type `UF_NEVER_DROP`.
+ * - all syscalls that are not managed through `GENERIC_EVENTS` and don't
+ *   have the `EF_DROP_SIMPLE_CONS` flag.
+ *
+ * Please note: if some syscalls miss probably you have an old kernel
+ * that don't define them. Try to use a newer one.
+ */
+
+void print_modern_probe_syscalls()
+{
+	char str[SYSCALL_TABLE_SIZE][SYSCALL_NAME_MAX_SIZE];
+	int interesting_syscall = 0;
+	enum ppm_syscall_code ppm_syscall_code = 0;
+
+	/* For every syscall of the system. */
+	for(int syscall_id = 0; syscall_id < SYSCALL_TABLE_SIZE; syscall_id++)
+	{
+		/* TAKE ALWAYS: If the syscall has `UF_NEVER_DROP` flag we cannot use simple consumer. */
+		if(g_syscall_table[syscall_id].flags & UF_NEVER_DROP)
+		{
+			ppm_syscall_code = g_syscall_code_routing_table[syscall_id];
+			if(!g_syscall_info_table[ppm_syscall_code].name)
+			{
+				printf("*ERROR: a `UF_NEVER_DROP` syscall has no the corresponding name in the g_syscall_info_table.\n");
+				exit(EXIT_FAILURE);
+			}
+			strcpy(str[interesting_syscall++], g_syscall_info_table[ppm_syscall_code].name);
+			continue;
+		}
+
+		/* TAKE NEVER: If we use generic events, we can drop the syscall with the simple consumer logic. */
+		if(g_syscall_table[syscall_id].enter_event_type == PPME_GENERIC_E)
+		{
+			continue;
+		}
+
+		ppm_syscall_code = g_syscall_code_routing_table[syscall_id];
+		if(g_syscall_info_table[ppm_syscall_code].flags & EF_DROP_SIMPLE_CONS)
+		{
+			continue;
+		}
+
+		/* This is an error since it means that a syscall we want to trace is not tracked in our `g_syscall_info_table`.
+		 * We have `EC_UNKNOWN` when we don't have an entry in the `g_syscall_info_table`.
+		 */
+		if(g_syscall_info_table[ppm_syscall_code].category == EC_UNKNOWN)
+		{
+			printf("*ERROR: a syscall that has an event associated is unknown in g_syscall_info_table.\n");
```

Comment:
Can we consider printing `g_syscall_info_table[ppm_syscall_code].name` along with the error message if it is not an empty string? I think it would make debugging this error message a lot easier.

Reply:
100% agree on this!

Reply:
Right now, there are no syscalls in the `g_syscall_info_table` that have both the `EC_UNKNOWN` flag and the `name`, and I think this is correct since if it is `EC_UNKNOWN` we don't know its name. `EC_UNKNOWN` flag is assigned by default to all entries that we don't define in the table because its value is `0`. Anyway, I can add this check even if i think that we will never have the `EC_UNKNOWN` plus the name

Reply:
Maybe we could print the `ppm_syscall_code` to understand which is the syscall involved. In this case, we will always have the information about the `ppm_code`. What do you think about that @Molter73 @FedeDP?

Reply:
If you feel the `ppm_syscall_code` is more useful than the name because it is not set when the category is unknown, then it is fine by me, my main concern here is having any sort of information to go on when this error is encountered 🙂

Reply:
Yes, I totally agree with you, I will add the `ppm_syscall_code` :smiley:. Thank you!